.github/workflows/main_check.yaml

```diff
@@ -0,0 +1,29 @@
+name: Pycloudlib Post Merge Check
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  post-merge-tests:
```

Comment:
We may be able to only trigger this job in CI by checking the whether we are on the upstream repo with the following:
```
on: [pull_request]
jobs:
  my-upstream-repo-test:
    if: github.event.pull_request.head.repo.full_name == github.repository
```
This would allow us (upstream pycloudlib commiters) to push our branches to upstream/myname/X and allow CI to run on all clouds for PRs we want to. And for PRs we don't want to run on,.... just push to our own remotes/forks.